src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -112,21 +112,28 @@ export const TradeConfirm = () => {
       // e.g sell asset AccountId, and sell asset address, and sell Txid
       // If we use the "real" (which we never get) buy Tx AccountId and address. then we'll never be able to lookup a Tx in state
       // and thus will never be able to react on the completed state
-      return serializeTxIndex(
-        sellAssetAccountId!,
-        buyTxid.toUpperCase(), // Midgard monkey patch Txid is lowercase, but we store Cosmos SDK Txs uppercase
-        fromAccountId(sellAssetAccountId!).account ?? '',
-      )
+
+      const thorOrderId = sellTradeId.toUpperCase()
+      const intoRune = isRune(trade?.buyAsset.assetId ?? '')
+      return intoRune
+        ? `${buyAssetAccountId}*${thorOrderId}*${trade?.receiveAddress}*OUT:${thorOrderId}`
```

Comment:
This might make sense to move into its own util, but since this is really a one-off thing for now encapsulated into `<TradeConfirm />`, totally fine as-is

Reply:
Yeah, if we ever use it again we absolutely should.